### PR TITLE
Use `BlockData` struct in `BlockAnalytics` trait

### PR DESCRIPTION
**Pull request by Alex6323**

### CODE REVIEW

**Comment by Alexandcoats** [comment]
What's with this history lol

File: src/analytics/tangle/block_activity.rs [hunk 1]
```diff
@@ -30,8 +29,8 @@ impl BlockAnalytics for BlockActivityAnalytics {
         self.measurement = BlockActivity::default();
     }
 
-    fn handle_block(&mut self, block: &Block, _: &[u8], _: &BlockMetadata) {
-        match block.payload {
+    fn handle_block(&mut self, block: &BlockData) {
```
Comment (Alexandcoats) [inline src/analytics/tangle/block_activity.rs:32]: Maybe destructure for simplicity?

**Comment by Alex6323** [comment]
> What's with this history lol

too lazy to create a new branch, sorry

File: src/tangle/sources/mod.rs [hunk 2]
```diff
@@ -24,18 +24,26 @@ use crate::types::{
 /// Logical grouping of data that belongs to a milestone.
 #[derive(Clone, Debug)]
 pub struct MilestoneData {
+    /// The [`MilestoneId`] of the milestone.
```
Comment (Alexandcoats) [inline src/tangle/sources/mod.rs:27]: These comments provide no benefit. I would rather have a `#[allow(missing_docs)]` tag

Comment (Alex6323) [reply src/tangle/sources/mod.rs:27]: fair enough ... (there they go ... my 5 mins of hard concentration to not make any typos)

File: src/analytics/tangle/block_activity.rs (hunk 1 above)
Comment (Alex6323) [reply src/analytics/tangle/block_activity.rs:32]: yup, good idea

**Review by grtlr — APPROVED** [review]

**Comment by grtlr** [comment]
I'm gonna squash it anyways.

**Review by grtlr — APPROVED** [review]

**Pull request merged**